src/Polly.Core/Outcome.TResult.cs

```diff
@@ -15,7 +15,13 @@ namespace Polly;
 public readonly struct Outcome<TResult>
 {
     internal Outcome(Exception exception)
-        : this() => ExceptionDispatchInfo = ExceptionDispatchInfo.Capture(Guard.NotNull(exception));
+        : this()
+    {
+        if (exception != null)
```

Comment:
The main problem here is that outcome that does not have exception nor result is basically an invalid instance.

> I had to do this so that it does not throw an error when the FaultGenerator doesn't return any exception to be injected.

Can fault generator return nullable `Outcome<T>?` (nullable)